app/src/main/java/com/github/geohunt/app/data/repository/bounties/TeamsRepository.kt

```diff
@@ -81,9 +81,13 @@ class TeamsRepository(
             val teamId = newTeamReference.key!!
 
             // set the leader
-            newTeamReference.child("name").setValue(name)
-            newTeamReference.child("score").setValue(0)
-            newTeamReference.child("teamLeader").setValue(teamLeaderUid)
+            // note that the three updates have to be atomic due to flows causing
+            // recomposition on partially created teams
```

Comment:
A small detail, but can be useful to have a more detail description of the bug when I won't review the code in a month because all of this is finished
```suggestion
            // Notice that the creation of teams should be atomic due to the UI
            // using flows that fetches every single changes made to the team 
            // reference. As such partially created teams causes for NPE
```

Reply:
Will add 👍